### PR TITLE
[stable/nginx-ingress] add HPA in case of Deployment

### PR DESCRIPTION
**Pull request by Fsero**
In case of deploying ingress controller as Deployment and configuring scaling rules, an HPA will be deployed to scale up and down the ingress controller.

### CODE REVIEW

File: stable/nginx-ingress/templates/controller-hpa.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{{- if eq .Values.controller.kind "Deployment" }}
+{{- if .Values.controller.autoscaling }}
```
Comment (chancez) [inline stable/nginx-ingress/templates/controller-hpa.yaml]: Feels like this should be based on a boolean like `.Values.controller.autoscaling.enabled`

Comment (Fsero) [reply stable/nginx-ingress/templates/controller-hpa.yaml]: changed!

**Comment by Fsero** [comment]
@chancez anything else I can do to merge this?

**Comment by chancez** [comment]
With HPA; do we need to worry about the changes to the deployment replica count via helm conflicting with what HPA is doing?

**Comment by Fsero** [comment]
not really, worst case scenario is HPA will add replica count set in Deployment with the minimum instances on HPA like explained here https://github.com/kubernetes/kubernetes/issues/25238

**Comment by chancez** [comment]
Seems fine to me. Someone with merge access will have to merge.

CC @mgoodness

**Pull request merged**